internal/build/phase_config_provider.go

```diff
@@ -108,11 +111,33 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 func WithDaemonAccess() PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
 		WithRoot()(provider)
+
 		bind := "/var/run/docker.sock:/var/run/docker.sock"
+
+		if dockerHost, ok := os.LookupEnv("DOCKER_HOST"); ok {
+			if strings.HasPrefix(dockerHost, "unix://") {
+				bind = fmt.Sprintf("%s:/var/run/docker.sock", strings.TrimPrefix(dockerHost, "unix://"))
+			}
+			if strings.HasPrefix(dockerHost, "tcp://") {
+				bind = ""
+				if runtime.GOOS == "linux" {
+					provider.hostConf.NetworkMode = "host"
+					provider.ctrConf.Env = append(provider.ctrConf.Env, fmt.Sprintf("DOCKER_HOST=%s", dockerHost))
```

Comment:
Yes, we probably should set the `provider.ctrConf.Env` in any case here. The thing is that there can be issue if the `TCP` destination is not reachable from internet, for instance if the `docker API` is running on localhost of the host. On Linux we can use `host` network, I don't know how to solve that on other supported `OS`es.

Reply:
> The proposed solution in this PR currently adds complexity that isn't quite a drop-in replacement. For example, if you use a `tcp://docker:1234` remote daemon and `docker` hostname is local to the client via `/etc/hosts` the lifecycle container wouldn't be able to resolve the hostname.

@jromero I know that's problematic, on Linux this can be solved by using `--network=host` I don't know how to fix that on other `OS`es.

Reply:
Maybe we could just use `os.Stderr` and not `errorWriter`.